### DISCUSSION TITLE
http2 requests

### DISCUSSION THREAD

**Discussion by J4CKVVH173** [actix/actix-web#3295] [Q&A] [answered]
Hi, I read that actix-web tries to open http 2 if it can. I have two apps on a server and i'd like to use an `actix client` to make http requests between my apps on the server but I can't find any option to say the client works in http 2.0. Do I need to make extra work to config the client or it works out of box in http 2 ? Do u know any ways to check type of connection?

**Comment by robjtede** [comment]
Connection type (HTTP 1 or 2) is a negotiation, on both sides, that is handled automatically when TLS (HTTPS) is used.

**Reply by J4CKVVH173** [reply]
@robjtede Do i have to set any extra headers or something like this? Or is the http/2 connection established automatically if it can? I am asking cause I know that I have to set in `nginx's` config extra option to work on http/2 but I can't any similar information about `actix-web` or `rewest` and also I don't know how to check connection type on the server side. If we speak about web browser I can check connection type in dev tools tab.

**Reply by robjtede** [reply]
The built-in [Logger](https://docs.rs/actix-web/latest/actix_web/middleware/struct.Logger.html) middleware will tell you what kind of connection is being used on each request.

> nginx's config extra option to work on http/2

Ahh, are you talking about using h2c (plaintext HTTP/2) as the backend connection?

If so, you're gonna want to set up the Actix Web server using automatic h2c detection using [`.bind_auto_h2c()`](https://docs.rs/actix-web/latest/actix_web/struct.HttpServer.html#method.bind_auto_h2c) instead of `.bind()`.

**Reply by J4CKVVH173** [reply]
@robjtede thanks for your advice with using Logger middleware.
When I mentioned nginx's config I was trying to show that if i want nginx works through http2 I have to explicitly specify this in the configuration file.
I am confused that in actix everything that I have to do it's set TLS connection and it will work in http2 :)

**Reply by robjtede** [reply, marked as answer]
> using h2c (plaintext HTTP/2) as the backend connection

Apologies, I've learned that nginx doesn't support this. Ignore this point.

I'm not sure I understand. What I can tell you is that when using HTTPS, Actix Web (and nginx / most web servers) automatically negotiates the connection type, h1 or h2 via [ALPN](https://en.wikipedia.org/wiki/Application-Layer_Protocol_Negotiation).

Does this solve your problem?

**Reply by J4CKVVH173** [reply]
@robjtede yeap, thanks :)
